x/ibc-transfer/handler_test.go

```diff
@@ -29,41 +28,38 @@ func (suite *HandlerTestSuite) SetupTest() {
 }
 
 // constructs a send from chainA to chainB on the established channel/connection
-// and sends the coins back from chainB to chainA.
-// FIX: this test currently passes because source is incorrectly determined
-// by the ibc-transfer module, so what actually occurs is chainA and chainB
-// send coins to each other, but no coins are ever sent back. This can be
-// fixed by receving and acknowledeging the send on the counterparty chain.
-// https://github.com/cosmos/cosmos-sdk/issues/6827
+// and sends the same coin back from chainB to chainA.
 func (suite *HandlerTestSuite) TestHandleMsgTransfer() {
 	clientA, clientB, _, _, channelA, channelB := suite.coordinator.Setup(suite.chainA, suite.chainB)
-	handlerA := ibctransfer.NewHandler(suite.chainA.App.TransferKeeper)
 
-	coinToSendToB := sdk.NewCoins(sdk.NewCoin(fmt.Sprintf("%s/%s/%s", channelB.PortID, channelB.ID, sdk.DefaultBondDenom), sdk.NewInt(100)))
+	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
 
 	// send from chainA to chainB
 	msg := types.NewMsgTransfer(channelA.PortID, channelA.ID, coinToSendToB, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), 110, 0)
 
-	res, err := handlerA(suite.chainA.GetContext(), msg)
-	suite.Require().NoError(err)
-	suite.Require().NotNil(res, "%+v", res) // successfully executed
-
-	err = suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, msg)
+	err := suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, msg)
 	suite.Require().NoError(err) // message committed
 
-	handlerB := ibctransfer.NewHandler(suite.chainB.App.TransferKeeper)
+	// relay send
+	fungibleTokenPacket := types.NewFungibleTokenPacketData(coinToSendToB.Denom, coinToSendToB.Amount.Uint64(), suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String())
+	packet := channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, 110, 0)
+	ack := types.FungibleTokenPacketAcknowledgement{Success: true}
+	err = suite.coordinator.RelayPacket(suite.chainA, suite.chainB, clientA, clientB, packet, ack.GetBytes())
+	suite.Require().NoError(err) // relay committed
```

Comment:
Can you also test that the receiver's balance on chainB has the vouchers with the denom and amount that you expect?